sites/all/modules/custom/practice/practice.module

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/**
+ * @file
+ * This is drupal 7 practice module.
+ */
+
+/**
+ * Represents max amount of latest nodes.
+ */
+define('PRACTICE_MAX_NODE_AMOUNT', 5);
+
+/**
+ * Represents max length of node title.
+ */
+define('PRACTICE_MAX_TITLE_LENGTH', 50);
+
+/**
+ * Implements hook_menu().
+ */
+function practice_menu() {
+  $items['admin/node-titles'] = array(
+    'title' => 'Latest node\'s titles',
+    'description' => 'Form with 5 latest node\'s titles',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('practice_latest_nodes_form'),
+    'access arguments' => array('access node tittle form'),
+    'type' => MENU_NORMAL_ITEM,
+  );
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function practice_permission() {
+  return array(
+    'access node title form' => array(
+      'title' => t('Access node tittle form'),
+      'description' => t('Access form with titles of latest 5 nodes.'),
+    ),
+  );
+}
+
+
+/**
+ * Builds latest nodes form.
+ */
+function practice_latest_nodes_form($form, &$form_state) {
+  $nids = array_keys(db_select('node', 'n')
+    ->fields('n', array('nid'))
+    ->orderBy('nid', 'DESC')
+    ->range(0, PRACTICE_MAX_NODE_AMOUNT)
+    ->execute()
+    ->fetchAllAssoc('nid'));
+
+  $nodes = node_load_multiple($nids);
+
+  if ($nodes) {
+    $form['nodes']['titles']['#tree'] = TRUE;
+    foreach ($nodes as $node) {
+      $form['nodes']['nid'][$node->nid] = array(
+        '#type' => 'html_tag',
+        '#tag' => 'h3',
+        '#value' => $node->nid,
+      );
+      $form['nodes']['titles'][$node->nid] = array(
+        '#type' => 'textfield',
+        '#default_value' => $node->title,
+      );
+      $form['nodes']['updated'][$node->nid] = array(
+        '#type' => 'html_tag',
+        '#tag' => 'h3',
+        '#value' => date('Y.m.d H:i:s', $node->changed),
+      );
+    }
+
+    $form['save_btn'] = array(
+      '#type' => 'submit',
+      '#value' => t('Save titles'),
+    );
+
+    $form['#theme'] = 'practice_form_table';
+    $form['#nids'] = serialize($nids);
+  }
+  else {
+    $form['warning'] = array(
+      '#type' => 'html_tag',
+      '#tag' => 'h2',
+      '#value' => t('Currently there are no nodes on the site'),
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * Performs validation for latest nodes form: checks title length.
+ */
+function practice_latest_nodes_form_validate($form, &$form_state) {
+  $titles = &$form_state['input']['titles'];
```

Comment:
Why are you passing $form_state by reference?

always use use $form_state['values'] instead of $form_state['input']
$form_state['input'] contains invalidated data
https://drupal.stackexchange.com/questions/112312/difference-between-form-statevalues-and-form-stateinput